blaze/compute/tests/test_sql_compute.py

```diff
@@ -1130,6 +1119,84 @@ def test_join_count():
             normalize(str(result)) == normalize(expected2))
 
 
+def test_transform_where():
+    t2 = t[t.id == 1]
+    expr = transform(t2, abs_amt=abs(t2.amount), sine=sin(t2.id))
+    result = normalize(str(compute(expr, s)))
+
+    expected = normalize("""SELECT
+        accounts.name,
+        accounts.amount,
+        accounts.id,
+        abs(accounts.amount) as abs_amt,
+        sin(accounts.id) as sine
+    FROM accounts
+    WHERE accounts.id = :id_1
+    """)
+
+    try:
+        # we're exactly equal
+        assert result == expected
+    except AssertionError:
+        # we're equal modulo additional transform columns because the **kwargs
+        # of transform are a dict and python functions don't have access to the
+        # order in which arguments were passed to the function
```

Comment:
Yep, I think we should sort so that we can get a predictable datashape out of a `Merge` expression